olive/systems/docker/docker_system.py

```diff
@@ -103,6 +108,69 @@ def __init__(
                     _print_docker_logs(e.build_log, logging.ERROR)
                     raise
 
+    def submit_workflow(self, run_config: RunConfig):
+        with tempfile.TemporaryDirectory() as tempdir:
+            self._run_workflow(run_config, tempdir)
+
+    def _ensure_directory(self, path: str, default: str) -> str:
+        dir_path = Path(path or default)
+        dir_path.mkdir(parents=True, exist_ok=True)
+        return str(dir_path)
```

Comment:
Recommend moving it to utils/common.py.

Also, the decision to use `path` vs. `default` should be caller's not this implementation. Within this implementation it could mean two different things -

* `mkdir(path)` if path is valid but doesn't exist
* `mkdir(default) if path is valid but it doesn't exist

The API will be cleaner if the caller explicitly passed one or the other.

If you like to keep the current implementation move it into the _run_workflow so it's invisible outside.

Reply:
This is internal use for docker system. But this may be removed in my follow-up PR. If run config is initialized and validated, it should have default value, so there will be no none check here.